storage/client/file/src/main/java/com/azure/storage/file/implementation/FilesImpl.java

```diff
@@ -67,68 +67,68 @@ public FilesImpl(AzureFileStorageImpl client) {
     }
 
     /**
-     * The interface defining all the services for Files to be used by the
-     * proxy service to perform REST calls.
+     * The interface defining all the services for AzureFileStorageFiles to be
+     * used by the proxy service to perform REST calls.
      */
     @Host("{url}")
-    @Service("Storage File Files")
+    @ServiceInterface(name = "AzureFileStorageFiles")
     private interface FilesService {
-        @PUT("{shareName}/{filePath}")
+        @Put("{shareName}/{directory}/{fileName}")
```

Comment:
The directory is not necessary for file client. Share is able to create file directly.

Reply:
yes I believe you can pass null to it.